### ISSUE TITLE
False positive on HTTPS detection

### ISSUE THREAD

**Issue by aprea** [woothemes/woocommerce#8479]
Had a problem spring up just recently on a few local & and live WooCommerce websites with regards to the following code:

```php
/**
 * Fix `$_SERVER` variables for various setups.
 *
 * Note: Removed IIS handling due to wp_fix_server_vars()
 *
 * @since 2.3
 */
private function fix_server_vars() {
	// NGINX Proxy
	if ( ! isset( $_SERVER['REMOTE_ADDR'] ) && isset( $_SERVER['HTTP_REMOTE_ADDR'] ) ) {
		$_SERVER['REMOTE_ADDR'] = $_SERVER['HTTP_REMOTE_ADDR'];
	}

	if ( ! isset( $_SERVER['HTTPS'] ) ) {
		if ( ! empty( $_SERVER['HTTP_HTTPS'] ) ) {
			$_SERVER['HTTPS'] = $_SERVER['HTTP_HTTPS'];
		} elseif ( ! empty( $_SERVER['HTTP_X_FORWARDED_PROTO'] ) && $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https' ) {
			$_SERVER['HTTPS'] = '1';
		}
	}
}
```

Which is located in `woocommerce.php`.

Both my sites are HTTP only, I don't have a SSL cert assigned to either the local or live site.

My WordPress Address and Site Address both use http://.

The problem occurs on the front end, all the assets are trying to load over HTTPS which is causing multiple 404s and preventing all the site's CSS, JS and image files from loading.

Calling `home_url()` during `init` also spits out a https address.

Note that this **only** occurs on Chrome, all other browsers appear to be working correctly.

Problem occurring on both nginx and apache.

**Comment by aprea** [comment]
Forgot to mention that it's definitely something to do with this function, if I add a `return;` to the top of the function everything works correctly in all browsers on the front end.

**Comment by aprea** [comment]
**more information** I just checked my Ubuntu box and wasn't able to replicate this behaviour.

So this appears to only affect Windows clients using Chrome.

**Comment by claudiosmweb** [comment]
We removed this from 2.4.
Will not be a problem soon.

**Comment by m9dfukc** [comment]
Hi,

I'm experiencing the same behaviour ... unfortunately this also seems to affect Safari 8.0.6 as well as Chrome Beta 44 (OsX version). This issue has also been reported on the [Chromium issue tracker](https://code.google.com/p/chromium/issues/detail?id=505268#c13).

**Comment by futuernorn** [comment]
Glad to see this is cleared up, and just to follow up on @m9dfukc's note, it is specifically an issue with the upcoming versions of chrome including a HTTPS header in its requests.

**Comment by svetto87** [comment]
Hello, I'm having the same issue after updating my chrome browser. First, I thought that my site is hacked, but then when I opened it in Firefox I saw that it's working fine.

Any tips on how to fix this issue on older versions of woocommerce?

**Comment by claudiosmweb** [comment]
@svetto87 fixed on WooCommerce 2.3.13

**Comment by DanielRuf** [comment]
@svetto87 you just need to apply the changes
https://github.com/woothemes/woocommerce/compare/2.3.11...release/2.3#diff-355207be481b8c53c1d92f0c86ab1086L382
